include/ignition/rendering/base/BaseCOMVisual.hh

```diff
@@ -0,0 +1,183 @@
+
+      /// \brief Get the radius of the CoM sphere
+      /// \return Radius of the CoM sphere
+      protected: double SphereRadius() const;
+
+      /// \brief Parent visual name.
+      protected: std::string parentName = "";
+
+      /// \brief Parent mass.
+      protected: double mass = 1.0;
+
+      /// \brief Inertia pose in parent frame.
+      protected: ignition::math::Pose3d inertiaPose =
+          ignition::math::Pose3d::Zero;
+
+      /// \brief Flag to indicate parent properties have changed.
+      protected: bool dirtyCOMVisual = false;
+    };
+
+    //////////////////////////////////////////////////
+    template <class T>
+    BaseCOMVisual<T>::BaseCOMVisual()
+    {
+    }
+
+    //////////////////////////////////////////////////
+    template <class T>
+    BaseCOMVisual<T>::~BaseCOMVisual()
+    {
+    }
+
+    /////////////////////////////////////////////////
+    template <class T>
+    void BaseCOMVisual<T>::PreRender()
+    {
+      T::PreRender();
+    }
+
+    //////////////////////////////////////////////////
+    template <class T>
+    void BaseCOMVisual<T>::Init()
+    {
+      T::Init();
+    }
+
+    //////////////////////////////////////////////////
+    template <class T>
+    void BaseCOMVisual<T>::SetInertial(
+          const ignition::math::Inertiald &_inertial)
+    {
+      this->inertiaPose = _inertial.Pose();
+
+      this->SetMass(_inertial.MassMatrix().Mass());
+    }
+
+    template <class T>
+    void BaseCOMVisual<T>::SetMass(double _mass)
+    {
+      if (_mass <= 0)
+      {
+        // Unrealistic mass, load with default mass
+        if (_mass < 0)
+        {
+          ignlog << "The parent " << this->parentName
+              << " has unrealistic mass, "
+              << "unable to visualize sphere of equivalent mass.\n";
+        }
+        else
+        {
+          ignlog << "The parent " << this->parentName
+              << " is static or has mass of 0, "
+              << "so a sphere of equivalent mass will not be shown.\n";
+        }
+        return;
+      }
+
+      this->mass = _mass;
+      this->dirtyCOMVisual = true;
+    }
+
+    //////////////////////////////////////////////////
+    template <class T>
+    double BaseCOMVisual<T>::Mass() const
+    {
+      return this->mass;
+    }
+
+    //////////////////////////////////////////////////
+    template <class T>
+    ignition::math::Pose3d BaseCOMVisual<T>::InertiaPose() const
+    {
+      return this->inertiaPose;
+    }
+
+    //////////////////////////////////////////////////
+    template <class T>
+    VisualPtr BaseCOMVisual<T>::SphereVisual() const
+    {
+      return nullptr;
+    }
+
+    //////////////////////////////////////////////////
+    template <class T>
+    double BaseCOMVisual<T>::SphereRadius() const
+    {
+      // Compute radius of sphere with density of lead and equivalent mass.
+      double sphereRadius;
+      double dLead = 11340;
```

Comment:
nit: use more verbose variable name `densityLead`

Reply:
Changed in [`aee5617`](https://github.com/ignitionrobotics/ign-rendering/pull/345/commits/aee5617f0745a06486c3c2bd95dc7f8a9a71763e)